great_expectations/data_context/data_context.py

```diff
@@ -110,83 +93,124 @@ def create(cls, project_root_dir=None):
         """DataContext constructor
 
         Args:
             context_root_dir: location to look for the ``great_expectations.yml`` file. If None, searches for the file \
             based on conventions for project subdirectories.
-            expectation_explorer: If True, load the expectation explorer manager, which will modify GE return objects \
-            to include ipython notebook widgets.
             data_asset_name_delimiter: the delimiter character to use when parsing data_asset_name parameters. \
             Defaults to '/'
 
         Returns:
             None
         """
-        self._expectation_explorer = expectation_explorer
-        self._datasources = {}
-        if expectation_explorer:
-            from great_expectations.jupyter_ux.expectation_explorer import ExpectationExplorer
-            self._expectation_explorer_manager = ExpectationExplorer()
-
-        # determine the "context root directory" - this is the parent of "great_expectations" dir
-        if context_root_dir is None:
-            if os.path.isdir("../notebooks") and os.path.isfile("../great_expectations.yml"):
-                context_root_dir = "../"
-            elif os.path.isdir("./great_expectations") and \
-                    os.path.isfile("./great_expectations/great_expectations.yml"):
-                context_root_dir = "./great_expectations"
-            elif os.path.isdir("./") and os.path.isfile("./great_expectations.yml"):
-                context_root_dir = "./"
-            else:
-                raise DataContextError(
-                    "Unable to locate context root directory. Please provide a directory name."
-                )
+        assert isinstance(project_config, DataContextConfig)
 
+        self._project_config = project_config
         self._context_root_directory = os.path.abspath(context_root_dir)
 
-        # TODO: these paths should be configurable
-        self.fixtures_validations_directory = os.path.join(self.root_directory, "fixtures/validations")
-        self.data_doc_directory = os.path.join(self.root_directory, "uncommitted/documentation")
-
-        self._project_config = self._load_project_config()
+        self._datasources = {}
 
         if not self._project_config.get("datasources"):
             self._project_config["datasources"] = {}
         for datasource in self._project_config["datasources"].keys():
             self.get_datasource(datasource)
 
-        plugins_directory = self._project_config.get("plugins_directory", "plugins/")
+        # self._plugins_directory = self._project_config.get("plugins_directory", "plugins/")
+        plugins_directory = self._project_config.get("plugins_directory")
+        #TODO: This check should not be necessary
+        if plugins_directory == None:
+            plugins_directory =  "plugins/"
+        #TODO: This should be factored out into a _convert_to_absolute_path_for_internal_use method, which should live in the @property, not here
```

Comment:
This was done in a later PR.

And to be precise, I believe this should be "before we merge this PR to develop". The whole point of the v0.8.0_prep branch is to allow us to address these kinds of things in multiple branch-merge operations before merging to develop.

Reply:
The review challenge here is that the TODOs are written as multi-part imperatives. I don't want to get into a situation where because an undone TODO on an unmerged branch wasn't objected to, that is now the assumed shared state of the whole team. It is a way of dramatically increasing the effective scope of a PR without even making the changes.